File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConstants.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3;

import io.airbyte.integrations.destination.s3.util.CompressionType;
import io.airbyte.integrations.destination.s3.util.S3NameTransformer;

public final class S3DestinationConstants {

  public static final String YYYY_MM_DD_FORMAT_STRING = "yyyy_MM_dd";
  public static final S3NameTransformer NAME_TRANSFORMER = new S3NameTransformer();
  public static final String DEFAULT_PATH_FORMAT = "${NAMESPACE}/${STREAM_NAME}/${YEAR}_${MONTH}_${DAY}_${EPOCH}_";

  // gzip compression for CSV and JSONL
  public static final String COMPRESSION_ARG_NAME = "compression";
  public static final String COMPRESSION_TYPE_ARG_NAME = "compression_type";
  public static final String FLATTEN_DATA = "flatten_data";
  public static final CompressionType DEFAULT_COMPRESSION_TYPE = CompressionType.GZIP;

  private S3DestinationConstants() {}

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3DestinationConstants.kt
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.integrations.destination.s3

import io.airbyte.cdk.integrations.destination.s3.util.CompressionType
import io.airbyte.cdk.integrations.destination.s3.util.S3NameTransformer

class S3DestinationConstants {

    companion object {
        const val YYYY_MM_DD_FORMAT_STRING: String = "yyyy_MM_dd"
        @JvmField val NAME_TRANSFORMER: S3NameTransformer = S3NameTransformer()
        const val DEFAULT_PATH_FORMAT: String =
            "\${NAMESPACE}/\${STREAM_NAME}/\${YEAR}_\${MONTH}_\${DAY}_\${EPOCH}_"

        // gzip compression for CSV and JSONL
        const val COMPRESSION_ARG_NAME: String = "compression"
        const val COMPRESSION_TYPE_ARG_NAME: String = "compression_type"
        @JvmField val DEFAULT_COMPRESSION_TYPE: CompressionType = CompressionType.GZIP

        // Flattening for CSV and JSONL
        const val FLATTENING_ARG_NAME: String = "flattening"
    }
}
